wpiutil/src/main/native/cpp/FileLogger.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#include "wpi/FileLogger.h"
+
+namespace wpi {
+FileLogger::FileLogger(std::string_view file,
+                       std::function<void(std::string_view)> callback)
+#ifdef __linux__
+    : m_fileHandle{open(file.data(), O_RDONLY)},
+      m_inotifyHandle{inotify_init()},
+      m_inotifyWatchHandle{
+          inotify_add_watch(m_inotifyHandle, file.data(), IN_MODIFY)},
+      m_thread{[=, this] {
+        char buf[4000];
+        struct inotify_event ev;
+        int len = 0;
+        while ((len = read(m_inotifyHandle, &ev, sizeof(ev))) > 0) {
+          int bufLen = 0;
+          if ((bufLen = read(m_fileHandle, buf, sizeof(buf)) > 0)) {
+            callback(std::string_view{buf, static_cast<size_t>(bufLen)});
+          }
+        }
+      }}
+#endif
+{
+}
+FileLogger::FileLogger(FileLogger&& other)
+#ifdef __linux__
+    : m_fileHandle{std::move(other.m_fileHandle)},
+      m_inotifyHandle{std::move(other.m_inotifyHandle)},
+      m_inotifyWatchHandle{std::move(other.m_inotifyWatchHandle)},
```

Comment:
`other`'s handles need to be invalidated:
```suggestion
    : m_fileHandle{std::exchange(other.m_fileHandle, -1)},
      m_inotifyHandle{std::exchange(other.m_inotifyHandle, -1)},
      m_inotifyWatchHandle{std::exchange(other.m_inotifyWatchHandle, -1)},
```